core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -841,8 +841,7 @@ public SegmentFile getSegmentFile() {
         }
         if (entry.getValue().status.equals(SegmentStatus.SUCCESS.getMessage())) {
           for (String indexFile : entry.getValue().getFiles()) {
-            indexFiles.put(location + CarbonCommonConstants.FILE_SEPARATOR + indexFile,
-                entry.getValue().mergeFileName);
+            indexFiles.put(location + CarbonCommonConstants.FILE_SEPARATOR + indexFile, null);
```

Comment:
Its caller `SegmentFileStore.getIndexCarbonFiles()`has code to handle for non null value to add merge index file also to index file list. It becomes dead code now. You would want to remove that too.

Reply:
This make sense, will do it.

Reply:
Done.